### PR TITLE
feat: US-6.2 cover mode (restyle a clip in a different genre)

### LINKED ISSUE #34: US-6.2: Cover mode (restyle a clip in a different genre)
## User Story

**As a** musician, **I want** to create a cover version of a clip in a different style, **so that** I can explore how a song sounds in another genre.

## Description

Preserves the melodic structure of the source while restyling instrumentation, arrangement, and optionally the vocal performance.

## Functional Requirements

- `acemusic cover <clip_id> --style "jazz piano trio"`
- `--voice <voice_id>` optional custom voice (Stage 25 prerequisite — implement as no-op or placeholder for now)
- `--lyrics "new lyrics"` optional lyrics override (melody preserved, words changed)
- Source clip's melody contour is retained in the output
- New clip created with `generation_mode: cover`

## Acceptance Criteria

- [ ] Cover output is recognizably derived from the source melody
- [ ] Style is audibly different from the original
- [ ] Lyrics override changes the words while keeping the melodic feel

## Dependencies

**Blocked by:** Stage 5 completion. Can be implemented in parallel with US-6.1.

## References

- User stories: `user-stories/01-layer-1-cli-foundation.md` — Stage 6, US-6.2

### PR DESCRIPTION
**Pull request by frankbria**
## Summary

- Adds `acemusic cover <clip_id> --style "..."` CLI command that creates a cover of an existing clip in a different style.
- Submits `task_type=cover` to ACE-Step with the source clip as `src_audio_path`; output is registered as a new clip with `parent_clip_id` set to the source and `generation_mode='cover'`.
- `--lyrics` overrides the source lyrics; `--voice` is a Stage-25 placeholder that prints an informational message.

Closes #34

## Implementation notes

- The existing `AceStepClient.submit_task()` already supports `task_type="cover"` and `src_audio_path` (from US-6.1), so no client changes were required.
- The command takes a `clip_id` (per US-6.2 spec) and follows the `extend` command pattern: DB lookup → submit task → poll → download → register new clip.
- Server-side path is used; ACE-Step must run on the same host or with shared filesystem access (matches the constraint already documented for `extend`).

## Test plan

- [x] 13 unit tests in `tests/test_cover.py` pass (mocked client + validation)
- [x] `@pytest.mark.integration test_cover_live_server` added (gated behind integration marker)
- [x] Full suite (`uv run pytest`) — 463 passed, 1 skipped
- [x] `uv run ruff check src/ tests/` — clean
- [x] `uv run black --check src/ tests/` — clean
- [x] `uv run acemusic cover --help` renders correct usage

## Summary by CodeRabbit

* **New Features**
  * Added "Cover Mode" CLI to create a cover version of an existing audio clip with selectable style, optional lyrics override, output/name options, and a placeholder notice for voice.

* **Documentation**
  * Added usage docs and a spec for Cover Mode, including CLI examples, options, acceptance criteria, polling/timeout, and error-handling guidance.

* **Tests**
  * Added comprehensive unit and integration tests covering success paths, validations, API interaction, polling, output handling, and error cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/frankbria/auto-music-studio/pull/60)

### CODE REVIEW

**Comment by frankbria** [comment]
Thanks for the reviews. Addressed in e3e3f16:

**Fixed:**
- Dropped the `prompt = style` alias; `style` is now passed directly as `prompt`.
- `--name` now sets both the filename slug and the saved clip title (previously only the filename, which was surprising).
- Resolved `src_wav` in `test_submits_cover_task_with_correct_params` to match the implementation's `.resolve()` call.
- Added `test_output_directory_overrides_default` and `test_name_option_sets_filename_and_title`.

**Deferred (out of scope for this PR):**
- **Extracting a shared `_poll_and_download` helper** — agreed this is the right refactor with 3 commands now sharing the pattern, but touching `extend` here would expand the PR's surface area significantly. Filing as a follow-up refactor PR so this one can ship focused on US-6.2.
- **`warnings.warn` → `console.print`** — kept for consistency with the `extend` pattern (US-6.1). Worth normalizing in the same follow-up refactor that extracts the polling helper.
- **Additional edge-case tests** (poll timeout, download error, empty audio_urls, db error) — these paths are exercised in `extend`'s test suite via the same code paths; adding them here would be largely copy-paste. Happy to add if you'd prefer.
- **Integration test source clip metadata** — left as null intentionally to verify the cover command tolerates minimal source metadata. Will add a comment to that effect in the follow-up.

**Comment by frankbria** [comment]
Round 2 review feedback addressed in 0f5e837:

**Fixed (correctness):**
- **Source duration guard** (CodeRabbit Major): cover now probes via `get_duration()` when `source.duration` is None, and exits 1 with a clear error if the probe fails. Prevents `audio_duration=None` from reaching ACE-Step.
- **Non-audio source rejection**: cover now checks `src_path.suffix` against `SUPPORTED_FORMATS` and refuses MIDI/text/other non-audio sources up front with a clear error.
- **Stop inheriting `model` / `inference_steps`**: the cover request doesn't send these parameters, so persisting them on the new clip misrepresented provenance. Both removed; `seed`, `bpm`, `key`, `vocal_language`, `style_tags`, `lyrics` are still inherited (those are either sent or directly relevant).

**Fixed (docs/tests):**
- Markdown code fence languages added in `demo-us-6.2.md` (MD040).
- Integration test source clip now has representative `bpm=120`, `key="C major"`, `seed=42`, `style_tags="ambient"`.
- Inline comment added explaining `result: dict = {}` init.

**New tests:**
- `test_non_audio_source_returns_error` — MIDI file → exit 1 with audio-format error
- `test_missing_duration_returns_error` — `duration=None` + probe raises → exit 1
- `test_missing_duration_falls_back_to_probe` — `duration=None` + probe succeeds → submit_task called with valid duration

Full suite: **468 passed, 1 skipped**. 18 unit tests in `tests/test_cover.py`.

**Still deferred (out of scope, will follow up):**
- Polling helper extraction (touches `extend` too)
- `warnings.warn` → `console.print` (matches `extend` pattern; same follow-up)
- `workspace_with_clip` fixture parameterization

**Comment by frankbria** [comment]
Round 4 review (84f86d1):

**Reconciled the demo/code conflict on `model` inheritance:**

CodeRabbit's earlier review correctly flagged that inheriting `source.model` and `source.inference_steps` on the cover clip overstates provenance — the cover request doesn't send those parameters to ACE-Step. I removed them from the `Clip` constructor in 0f5e837 but the demo doc still claimed model was inherited. Demo doc now explicitly documents the *intentional* omission and the reason.

**Added timeout test:**
- `test_poll_timeout_returns_error` — sets `ACEMUSIC_POLL_TIMEOUT=0`, asserts exit 1 and "timed out" in output.

Full suite: 19 unit tests pass in `tests/test_cover.py`.

File: demo-us-6.2.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+```bash
+$ uv run pytest tests/test_cover.py
+collected 14 items / 1 deselected / 13 selected
+
+tests/test_cover.py .............    [100%]
+13 passed, 1 deselected
+```
+
+Full suite: **463 passed, 1 skipped**.
```
[the remark this answers is not in the document]
Comment (frankbria) [reply demo-us-6.2.md]: Already resolved in commit 04ab220 — the demo doc currently shows 19 selected, 19 passed, 1 deselected, and full suite 469 passed, 1 skipped, which matches the actual test collection output (19/20 collected, 1 deselected). The numbers CodeRabbit references are from an earlier diff iteration.
